prune.go

```diff
@@ -100,6 +100,41 @@ func shouldPrune(oldCount, newCount int, pruneOptions PruneOptions) (bool, strin
 	return false, "transactions have not grown significantly"
 }
 
+// checkMongoSupportsOut verifies that Mongo supports "$out" in an aggregation
+// pipeline. This was introduced in Mongo 2.6
+// https://docs.mongodb.com/manual/reference/operator/aggregation/out/
+func checkMongoSupportsOut(db *mgo.Database) bool {
+	var dbInfo struct {
+		VersionArray []int `bson:"versionArray"`
+	}
+	if err := db.Run(bson.M{"buildInfo": 1}, &dbInfo); err != nil {
+		return false
+	}
+	logger.Debugf("buildInfo reported: %v", dbInfo.VersionArray)
+	if len(dbInfo.VersionArray) < 2 {
+		return false
+	}
+	// Check if we are < 2.6
+	if dbInfo.VersionArray[0] < 2 ||
+		(dbInfo.VersionArray[0] == 2 && dbInfo.VersionArray[1] < 6) {
+		return false
+	}
+	return true
+}
+
+
+func getOracle(db *mgo.Database, txns *mgo.Collection, txnsCount int, maxMemoryTxns int) Oracle {
+	if txnsCount < maxMemoryTxns {
+		return NewMemOracle(txns)
+	}
+	// We are using Mongo 3.2 with a large txn array, use the db-based
```

Comment:
Cleaned up the comments, added a reference to:
 https://github.com/juju/mgopurge/issues/8